pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -21,30 +21,62 @@ router.get('/', function(req, res, next) {
     sqldb.query(sql.select_assessment_instances, params, function(err, result) {
         if (ERR(err, next)) return;
         res.locals.user_scores = result.rows;
+        
+        let time_limit_list = new Object();
+        let remaining_time_min = null;
+        let remaining_time_max = null;
+        let has_open_instance = false;
+        let has_closed_instance = false;
+        result.rows.forEach(function(row) {
+            if (!row.open) {
+                has_closed_instance = true;
+            } else if (row.time_remaining_sec === null) {
+                has_open_instance = true;
+            } else {
+                if (!(row.total_time_sec in time_limit_list)) {
+                    time_limit_list[row.total_time_sec] = row.total_time;
+                }
+                if (remaining_time_min === null ||
+                    remaining_time_min > row.time_remaining_sec) {
+                    remaining_time_min = row.time_remaining_sec;
+                }
+                if (remaining_time_max === null ||
+                    remaining_time_max < row.time_remaining_sec) {
+                    remaining_time_max = row.time_remaining_sec;
+                }
+            }
+        });
+        time_limit_list = Object.values(time_limit_list);
+        if (time_limit_list.length > 5) {
+            time_limit_list.splice(3, time_limit_list.length - 4, '...');
+        }
+
+        res.locals.time_limit_totals = {
+            total_time: time_limit_list.length > 0 ? time_limit_list.join(', ') : 'No time limits',
+            time_remaining_sec: remaining_time_max,
+            has_open_instance: has_open_instance,
+            has_closed_instance: has_closed_instance,
+            action: 'set_time_limit_all',
+        };
+        if (remaining_time_min === null)
+            res.locals.time_limit_totals.time_remaining = 'No time limits';
+        else if (remaining_time_max < 60)
+            res.locals.time_limit_totals.time_remaining = 'Less than a minute';
+        else if (remaining_time_min < 60)
+            res.locals.time_limit_totals.time_remaining = 'up to ' + Math.floor(remaining_time_max / 60) + ' min';
+        else if (Math.floor(remaining_time_min / 60) == Math.floor(remaining_time_max / 60))
+            res.locals.time_limit_totals.time_remaining = Math.floor(remaining_time_min / 60) + ' min';
+        else
+            res.locals.time_limit_totals.time_remaining = 'between ' + Math.floor(remaining_time_min / 60) + ' and ' + Math.floor(remaining_time_max / 60) + ' min';
```

Comment:
Nit: curly braces.
```suggestion
        if (remaining_time_min === null) {
            res.locals.time_limit_totals.time_remaining = 'No time limits';
        } else if (remaining_time_max < 60) {
            res.locals.time_limit_totals.time_remaining = 'Less than a minute';
        } else if (remaining_time_min < 60) {
            res.locals.time_limit_totals.time_remaining = 'up to ' + Math.floor(remaining_time_max / 60) + ' min';
        } else if (Math.floor(remaining_time_min / 60) == Math.floor(remaining_time_max / 60)) {
            res.locals.time_limit_totals.time_remaining = Math.floor(remaining_time_min / 60) + ' min';
        } else {
            res.locals.time_limit_totals.time_remaining = 'between ' + Math.floor(remaining_time_min / 60) + ' and ' + Math.floor(remaining_time_max / 60) + ' min';
        }
```

Reply:
I'd prefer not to use braces here, since there's a single variable being modified based on these rules, but I don't have a strong preference.